File: leetcode/1344. 时钟指针的夹角.go
func angleClock(hour int, minutes int) float64 {
	const oneMinuteAngle = 6.0
	const oneHourAngle = 30.0

	// 分针角度
	minuteAngle := float64(minutes) * oneMinuteAngle

	// 时针角度：基础小时角度 + 分钟带来的偏移
	hourAngle := float64(hour%12)*oneHourAngle + float64(minutes)*(oneHourAngle/60.0)

	// 计算夹角差值的绝对值
	diff := math.Abs(hourAngle - minuteAngle)

	// 返回最小夹角
	if diff < 360.0-diff {
		return diff
	}
	return 360.0 - diff
}

File: leetcode/1344. 时钟指针的夹角.rs
impl Solution {
    pub fn angle_clock(hour: i32, minutes: i32) -> f64 {
        let one_minute_angle = 6;
        let one_hour_angle = 30;
        let minute_angle = minutes as f64 * one_minute_angle as f64;
        let hour_angle = ((hour % 12) as f64 * one_hour_angle as f64)
            + (minutes as f64 * (one_hour_angle as f64 / 60.0));
        let mut diff = hour_angle - minute_angle;
        diff = diff.abs();
        diff.min(360.0 - diff)
    }
}